### PR TITLE
fix: timezone adjustment

### PR DESCRIPTION
**Pull request by ColinBuyck**
# Pull Request Template

## Issue Overview

This PR addresses #3518 

- [ ] This change addresses the issue in full
- [x] This change addresses only certain aspects of the issue (needs to be pulled into hba)
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

For this fix, I followed the pattern established by this [Detroit PR](https://github.com/CityOfDetroit/bloom/pull/1612) so that the UTC time would be converted to the exporters local timezone. Only additional changes, made in relation to manual-entry paper applications where the timing is manually entered and isn't affected by timezone adjustments of the db so the previous logic is maintained for those pieces. 

## How Can This Be Tested/Reviewed?

The most challenging component to the PR proved to be the replicating the DB's responses and time storage locally. Despite changing my databases timezone to UTC, I could not repeat this issue locally. However, the two screenshots below outline the DB responses seen on dev and I **added testing that directly mirrors these responses** to demonstrate this approach's effectiveness. 

Screenshot 2023-06-27 at 8 04 47 AM
Screenshot 2023-06-27 at 8 04 19 AM

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: sites/partners/src/lib/hooks.ts [hunk 1]
```diff
@@ -499,9 +499,10 @@ export const createDateStringFromNow = (format = "YYYY-MM-DD_HH:mm:ss"): string
 
 export const useApplicationsExport = (listingId: string, includeDemographics: boolean) => {
   const { applicationsService } = useContext(AuthContext)
+  const timeZone = Intl.DateTimeFormat().resolvedOptions().timeZone.replace("/", "-")
```
Comment (ludtkemorgan) [inline sites/partners/src/lib/hooks.ts:502]: When testing this locally the time on the export is different than the one on the UI. Specifically the export is in Central time (the timezone my browser is in) whereas the UI shows pacific time (I believe that is hard coded in the code)

Comment (ColinBuyck) [reply sites/partners/src/lib/hooks.ts:502]: Good catch! My most recent changes should resolve that issue ⌚

File: backend/core/test/lib/format-local-date.spec.ts [hunk 2]
```diff
@@ -11,9 +11,14 @@ describe("formatLocalDate", () => {
       "04-01-2023 05:00:00PM"
     )
   })
-  test("with a format and timezone", () => {
+  test("with a format and EDT timezone", () => {
```
Comment (emilyjablonski) [inline backend/core/test/lib/format-local-date.spec.ts:14]: Great tests!

**Review by emilyjablonski — APPROVED** [review]

File: backend/core/src/applications/helpers.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import { ApplicationSubmissionType } from "../../types"
+import { isEmpty } from "class-validator"
+
+export const formatApplicationDate = (
+  submissionType: ApplicationSubmissionType,
+  dateString: string,
+  timeZone?: string
+) => {
+  if (isEmpty(dateString)) return ""
+  if (submissionType === "electronical") {
```
[the remark this answers is not in the document]
Comment (ColinBuyck) [reply backend/core/src/applications/helpers.ts:12]: That makes sense, thank you!

**Pull request merged**